client/main.go

```diff
@@ -589,41 +636,24 @@ func DoGet(remoteObject string, localDestination string, recursive bool) (transf
 		return nil, err
 	}
 	remoteObjectUrl.Scheme = remoteObjectScheme
-	fd := config.GetFederation()
-	defer config.SetFederation(fd)
-
-	// If there is a host specified, prepend it to the path in the osdf case
-	if remoteObjectUrl.Host != "" {
-		if remoteObjectUrl.Scheme == "osdf" {
-			remoteObjectUrl.Path, err = url.JoinPath(remoteObjectUrl.Host, remoteObjectUrl.Path)
-			if err != nil {
-				log.Errorln("Failed to join source url path:", err)
-				return nil, err
-			}
-		} else if remoteObjectUrl.Scheme == "pelican" {
-
-			config.SetFederation(config.FederationDiscovery{})
-			federationUrl, _ := url.Parse(remoteObjectUrl.String())
-			federationUrl.Scheme = "https"
-			federationUrl.Path = ""
-			viper.Set("Federation.DiscoveryUrl", federationUrl.String())
-			err = config.DiscoverFederation()
-			if err != nil {
-				return nil, err
-			}
-		}
-	}
 
+	// This is for condor cases:
```

Comment:
Just like you've factored out URL processing for metadata, you can factor out this logic for unsupported schemes. Will make a good unit test too.